src/main/webapp/app/entities/programming-exercise/programming-exercise-instruction.component.ts

```diff
@@ -193,6 +215,25 @@ export class ProgrammingExerciseInstructionComponent implements OnChanges, OnDes
         });
     }
 
+    /**
+     * Add task icons (success or failed) to tasks in introduction file.
+     * Existing icons will be removed.
+     */
+    private setUpTaskIcons() {
+        this.steps.forEach(({ done }, i) => {
```

Comment:
As we don't use the fa icons stylesheet (https://fontawesome.com/start), we can't directly use the icons in html.
Instead an Angular component needs to be created.
With the componentFactoryResolver an Angular component can be instantiated and inserted into the html manually.